### DISCUSSION TITLE
Is it me or does Expo Go client keep crashing on start in iOS Simulator? (M1)

### DISCUSSION THREAD

**Discussion by FaisalBinAhmed** [expo/expo#17811] [Archived: Questions] [answered]
Hi,
Ever since I got this new M1 Macbook Pro, I was unable to run my iOS RN app inside the Expo Go client in iOS Simulator. This keeps happening no matter what I try. 

I have tried reinstalling Xcode, Expo CLI, and Node. Updating Xcode, and Expo to the latest version. This happens in a brand new, just bootstrapped expo-typescript project. I have also tried resetting the simulator(s) and reinstalling the Expo Go client inside it.

Here is the [screen record](https://user-images.githubusercontent.com/20614782/173098139-588acd2d-fe02-4f9d-addc-52afcce8d81a.mp4) when I press `i` to open it in the iOS Simulator from the command line. The Expo Go app keeps crashing on its own.

What should I do at the moment? I don't want to install Rosetta for this.

System info:
OS: macOS 12.2.1
Rosetta: No Rosetta
Node: 16.15.1
Expo-cli: 5.4.9
Xcode: 13.4.1

**Comment by michael-gates-techngs** [comment]
Not an answer: 

Not sure if it's related, but it's crashing on start for me as well but on the Android simulator. Also on Expo 5.4.9 and Node 16.15.1, using an M1 Macbook on 12.4 Monterrey.

**Comment by huseyn0w** [comment]
Have the exact same issue. Anybody resolved it?

**Comment by huseyn0w** [comment]
@FaisalBinAhmed just figured out the reason in the documentation. on M1 Monterey it will not run without Rosetta as it is Intel app

**Reply by FaisalBinAhmed** [reply]
I don't understand the being Intel only binary part. Which do you mean is the Intel binary? Expo CLI tool? The simulator? Both run fine on their own without Rosetta.

**Reply by huseyn0w** [reply]
@FaisalBinAhmed I'm talking about Expo Go app, which is working on Intel chips, not Apple without Rosetta 2. it does not run for me on OS Monterey, unless I installed Rosetta 2, which enables Intel based apps to run on M1.

**Reply by FaisalBinAhmed** [reply]
To be honest, I am a little confused by the expo go comment. Expo Go is written for iPhone, an ARM device already. Why would it require an Intel machine to run? On Intel machines it would be simulated; on ARM machines like M1 devices, it can run natively.

**Reply by brentvatne** [reply]
expo go works as expected on m1 and m2 (i currently use an m2 macbook air). it's possible that there is an issue with your simulator, you can try resetting it to default ([more info](https://stackoverflow.com/a/9088416/659988)).

if the issue persists, you can open up Console.app and find the device, press "Start" to see logs. share relevant logs here

**Reply by danrivett** [reply]
I'm also having this problem with SDK 46 on an M1 device without Rosetta 2 installed.

I am running the local expo-cli and have a `"expo": "^46.0.0"` dependency (which resolves to `46.0.2`) as well as:

```json
"resolutions": {
  "@expo/cli": "0.3.2"
}
```

Still the Expo Go app crashes on startup in the iOS Simulator. I don't want to install Rosetta 2 just for Expo Go to work in the iOS Simulator. Currently I am having to test on an older Macbook when I want to test on the simulator rather than on device.

**Reply by rdewolff** [reply]
This can be done instantly with the command:
```
/usr/sbin/softwareupdate --install-rosetta --agree-to-license
```
for anyone looking ^__^

**Comment by lucksp** [comment]
i just updated the expo version on a new app from scratch to `"expo": "^46.0.8"` from `44.0.6`.  Now when i build the Expo App on an iOS Simulator crashes.

My laptop is an MacBook Pro with:
- 12.5.1 (Monterey)
- 2.4 GHz 8-Core Intel Core i9
- 64 GB 2667 MHz DDR4

I have tried restarting laptop, updated my expo cli `yarn add global expo-cli@latest` which output "info" lines:
<img width="750" alt="image" src="https://user-images.githubusercontent.com/15162169/186967496-39ae215b-523d-49ff-8861-24e972e57701.png">

After this install, and running a clean `yarn` dep install and then `yarn start --clear i` this is the output:
<img width="1137" alt="image" src="https://user-images.githubusercontent.com/15162169/186974296-c48deca9-ef31-4ccb-9718-f660b9043669.png">

**Reply by brentvatne** [reply]
- we recommend using npm instead of yarn for global installs. yarn has some very strange global install behavior
- with sdk 46, you shouldn't be using the global expo-cli package anymore. [more info here](https://blog.expo.dev/the-new-expo-cli-f4250d8e34210), tl;dr: run `yarn expo start` or `npx expo start` in your project instead.

**Reply by lucksp** [reply]
can you repost the link?  getting a 404...

**Reply by lucksp** [reply]
> Hello @lucksp
> 
> I have the same issue with you.
> 
> Have you fixed the issue?
> 
> Looking forward to hearing from you.

This appears to have been related to an invalid custom `metro.config` file related to symlinking changes in the new SDK

**Comment by nuttyartist** [comment]
Same problem here.

EDIT: sorry, it happens for me not in the simulator but with the ios app.

**Comment by avoajaugochukwu** [comment]
Installing rosetta 2 is a good place to start.

**Comment by evanmazor** [comment]
Yup, I had this same problem and `softwareupdate --install-rosetta` fixed it

**Reply by dannyhw** [reply]
Wow... this actually solves it. Can your explain how you found this? Or why this is needed?

Thanks anyway.

**Reply by lukefernandez** [reply]
If that command doesn't quite work, which it didn't for me, try adding `--agree-to-license` like so:

`softwareupdate --install-rosetta --agree-to-license`

Doing that and restarting worked for me!

**Reply by pulkitgupta2** [reply]
This worked for me too!

**Reply by froilansam** [reply]
Wow. Worked!

**Reply by jeokeefe** [reply]
worked but i dont understand why

**Reply by sagardewani** [reply]
Restarting the simulator worked for me on M1 Pro

**Reply by ahujapiyush** [reply]
Thank you worked for me

**Reply by eagleeyejack** [reply]
Yep run `softwareupdate --install-rosetta` and restart simulator and you're off to the races

**Reply by tinhtony2506** [reply]
GREAT, worked for me

**Comment by MehmetKaplan** [comment]
Is it possible Expo Go app on simulator is compiled only on Intel chips? 
(Just imagining...) May be the compilation should be doubled for both Intel and Apple chips and the relevant one should be installed referring to the current system.

**Comment by UrielCh** [comment]
summery:
```bash
sudo xcode-select -s /Applications/Xcode.app
/usr/sbin/softwareupdate --install-rosetta --agree-to-license
```
fix all expo getting start issues on a new Mac.

the `xcode-select` fix the non-detection of Xcode.

**Comment by tsapeta** [comment, marked as answer]
So far Expo Go for simulator didn't support Apple Silicon because MLKit from Google that we used for `expo-face-detector` is still not supporting it. We've decided to remove that package in SDK 49, so this problem should go away once we release it 🙂

**Reply by FaisalBinAhmed** [reply]
Finally an answer for the original problem! Thanks for clarifying this.

**Reply by tsapeta** [reply]
Cool and sorry for the late response – we don't look at the discussions very often (we rather used Discord and Twitter), but I just yesterday subscribed to receive notifications from all discussions 🙈

**Reply by michelleabsantos** [reply]
I've been having this problem for 5 days now, my IOS simulator won't open. And I have to deliver my project next week. I'm so worried. I tried everything you can't think of, because in these 5 days I've been searching for solutions on many support channels, but no luck. 
@tsapeta would you please explain what I should do now to resolve the issue?

**Reply by tsapeta** [reply]
@michellebessa I think the solution mentioned here https://github.com/expo/expo/discussions/17811#discussioncomment-6088265 is a correct workaround to this issue.

**Comment by basafilm** [comment]
runtimeVersion: "48.0.0", for expo go

**Comment by jrmatos** [comment]
running `softwareupdate --install-rosetta` fixed the issue for me

**Comment by tsapeta** [comment]
The issue should be solved in SDK 49, so I'm closing the discussion.

**Reply by ahirdman** [reply]
I'm still having this issue with SDK 49.

I updated rosetta and application is running fine with deveolpment build.

Using a MacBook Pro with Apple M1 Pro

**Reply by tsapeta** [reply]
Sorry for confusion – I referred to just Expo Go. You may still have some libraries that doesn't work with Apple Silicon, such as `expo-face-detector` which depends on Google's MLKit that doesn't support it.
I would really appreciate if someone could provide a minimal project where I could reproduce this issue.

**Comment by brunoalexsander** [comment]
The same problem. I can't launch Expo Go on an iOS 16.4 Simulator.

Apple M2 Pro
SDK 49
npx create-expo-app AwesomeProject

**Reply by yolpsoftware** [reply]
Same here. Just upgraded my app to Expo SDK 49, and it keeps crashing in Expo Go on the simulator. Apple M2 Air.

Never mind, my crash was caused by a Reanimated worklet that called a static method of a class. Seems that static methods must not be called by worklets, even if they are marked as worklets themselves.

**Reply by brunoalexsander** [reply]
I created a new project through this command;

npx create-expo-app AwesomeProject --template blank@49

And everything seems to be working properly.

**Comment by robozb** [comment]
i experienced same error on real ipad

https://youtube.com/shorts/_EBdC-R3VRg?si=1R2Vfkgd6jU5lqV_

exit without any message
